### PR TITLE
Managing the Heap in C

### PR DESCRIPTION
**Pull request by botahamec**
Name: Managing the Heap in C
Description: Working with the heap in C is very difficult. Even Google struggles with it. But it's very important to not have memory issues in your code. This article will show you how to do that.

### CODE REVIEW

**Comment by hectorkambow** [comment]
Good afternoon @botahamec  - started a preliminary review on your recent article - 
Quick question - was there a style choice behind the sources in the intro paragraph one word after another(didn't want to mess it up if there was) - would it be possible to spread those out through the article - just from the audience point of view I wasn't aware those were different link to different sources each until I clicked through.

**Comment by botahamec** [comment]
@hectorkambow Yeah, that was purposeful. I took the idea from [my friend's article](https://www.thinkingmuchbetter.com/main/debugging-bad-detective/) on Thinking Much Better (where it says "browser of choice").

I can see your point. Although, I don't think I would spread it throughout the entire article. I don't want this to be an article all about how C is weird. These aren't sources, so it's not like I'm making separate points either. I could do something like this:

> You can use a language with better or easier memory management, like [Rust](www.rust-lang.org), [Java](www.java.com), [Python](www.python.org), or [Haskell](www.haskell.com).

This is fewer examples, but it still gets the point across. It loses the style though. A better compromise might be:

> You can use a language with [better](www.rust-lang.org) or [easier](www.python.org) memory management.

**Comment by botahamec** [comment]
Also, I noticed in the "minor edit" commit that the word "latter" was deleted, which doesn't make much sense. Now it says, "We'll be talking about the today"

**Comment by botahamec** [comment]
@hectorkambow I decided to go with the second option and fix both issues.

**Comment by hectorkambow** [comment]
> Also, I noticed in the "minor edit" commit that the word "latter" was deleted, which doesn't make much sense. Now it says, "We'll be talking about the today"

Yup - that was my edit - I think I was just reading the sentence weird. (got caught up with the links - forgot to go back) 
My mistake - will change it back before merging. 

Think I meant to change it to "we'll be talking about that today"

**Comment by hectorkambow** [comment]
> @hectorkambow I decided to go with the second option and fix both issues.

👍  thank you

**Comment by hectorkambow** [comment]
Good afternoon @interludic2000 - could you please give this article a tech review whenever you are free. Thank you 🚀

File: articles/managing-heap/index.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+author: mike-white
+date: 2020-07-16T00:00:00-07:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/managing-heap/hero.jpg
+    alt: memory drive image example
+---
+Memory management is very important. In Google Chrome, [70% of security bugs are memory problems](https://zdnet.com/article/chrom-70-of-all-security-bugs-are-memory-safety-issues/). Fixing memory management would lead to much more reliable and secure programs. There are two possible solutions to this. You can use a language with [better](https://www.rust-lang.org) or [easier](https://www.java.com), or just learn how to use memory properly. We will talk about the latter today.
```
Comment (interludic2000) [inline articles/managing-heap/index.md:17]: Link to https://zdnet.com/article/chrom-70-of-all-security-bugs-are-memory-safety-issues/ returns a 404

Comment (interludic2000) [reply articles/managing-heap/index.md:17]: > You can use a language with better or easier,

Feels like this sentence should read more along the lines of 
"You can use a language that manages memory better or easier"

Comment (botahamec) [reply articles/managing-heap/index.md]: I think I forgot an 'e' in the link. It seems to work now.

I think I just accidentally deleted the words, "memory management".

**Comment by botahamec** [comment]
Fixed

**Comment by interludic2000** [comment]
Nice one, approved.

**Pull request merged**